doc/whats_new/v0.24.rst

```diff
@@ -589,6 +589,10 @@ Changelog
 :mod:`sklearn.utils`
 ....................
 
+- |Enhancement| Avoid nonlinear complexity for ``repr`` of nested estimators
+  when `print_changed_only=True`.
```

Comment:
I think we should avoid the term nonlinear complexity and instead be very explicit that the calls are now faster.

Also, this should be instead in a `Miscellaneous` section as done in `doc/whats_new/v0.23.rst`

```suggestion
- |Enhancement| Calls to ``repr`` are now faster when `print_changed_only=True`, especially with meta-estimators.
```